### DISCUSSION TITLE
Recommended way to store channel names in `pynwb.Timeseries`

### DISCUSSION THREAD

**Discussion by etiennedemontalivet** [NeurodataWithoutBorders/helpdesk#68] [General]
Hi,

I could not find an example that shows how to store the channel names in `pynwb.TimeSeries`. I have seen there is no way to store it into the class, but I assume it's possible using one of the `data` type `(ndarray or list or tuple or Dataset or Array or StrDataset or HDMFDataset or AbstractDataChunkIterator or DataIO or TimeSeries)`.
I have tried with hdf5, which works, but data are not saved within nwb file, so this is not an option for me.

Could someone provide a simple example about how I can store channel names into `pynwb.TimeSeries` please ?

``` python
# for example, I have 3 channels
ecog_data = np.random.rand(10000, 3)
ch_names = ['ch_17', 'ch_12', 'ch_4']
# I want to save the data in a TimeSeries:
ecog_ts = pynwb.TimeSeries(
    name="ecog",
    description="ECoG data",
    data=ecog_data,
    unit="µV",
)
# how to store ch_names info ?
```

PS: I have used simple trick until now, such as storing them into the description, but I'm guessing there is a standard way as it is very common

Any help would be very much appreciated!

**Comment by CodyCBakerPhD** [comment]
Hi there @etiennedemontalivet

Generic `TimeSeries` don't have strict definitions on their axes after the first dimension (which should always be time), and so also don't support labeling of that kind

It sounds like what you actually want is the [`ElectricalSeries`](https://pynwb.readthedocs.io/en/stable/tutorials/domain/ecephys.html#raw-voltage-data) (which has an `electrodes` region reference as well as a lot of other channel specific metadata) - though take note that it fixes the `unit` to `Volts` so if your data is in `uV` you will need to apply a `conversion` factor scaled by `1e-6`

Check out the [ecephys tutorial](https://pynwb.readthedocs.io/en/stable/tutorials/domain/ecephys.html#extracellular-electrophysiology-data) for full examples of how to include all the metadata and write the data series

Just let us know if you have any more questions!

**Reply by etiennedemontalivet** [reply]
Thank you for pointing me in the right direction!
